backend/typescript/utilities/logger.ts

```diff
@@ -23,4 +23,31 @@ if (process.env.NODE_ENV !== "production") {
   );
 }
 
+export const logfactory = (filename: string): winston.Logger => {
+  const log = winston.createLogger({
+    level: "info",
+    format: winston.format.combine(
+      winston.format((info: winston.Logform.TransformableInfo) => {
+        const newInfo = {
+          level: info.level,
+          message: `${filename} ${info.message}`,
+        };
+        return newInfo;
+      })(),
+      winston.format.timestamp(),
+      winston.format.json(),
+    ),
+    transports: [
+      new winston.transports.File({ filename: "error.log", level: "error" }),
+      new winston.transports.File({ filename: "combined.log" }),
+    ],
```

Comment:
What do the level, format, and transports do?

Reply:
Transports are the files that you log the data to. Level is the the importance level of the log. For instance, we have normal info logs, and more high priority error logs. More information here: https://github.com/winstonjs/winston

This shouldn't be a big issue since the level & filename configurations were in the logger we had previously.

The format specifies the format the log is outputted in. The only new thing I added was a function to insert the filename into the log.

Reply:
cool, I was just curious about how it worked, thanks!